static/app/views/profiling/landing/chart.tsx

```diff
@@ -0,0 +1,184 @@
+import {axisLabelFormatter, tooltipFormatter} from 'sentry/utils/discover/charts';
+import getDynamicText from 'sentry/utils/getDynamicText';
+import {decodeScalar} from 'sentry/utils/queryString';
+
+interface Props extends WithRouterProps {
+  location: Location;
+  traces: Trace[];
+  end?: string;
+  start?: string;
+  statsPeriod?: string | null;
+  utc?: string;
+}
+
+function _ProfileScatterChart({
+  router,
+  location,
+  traces,
+  start,
+  end,
+  statsPeriod,
+  utc,
+}: Props) {
+  const theme = useTheme();
+
+  const colorEncoding = useMemo(() => getColorEncodingFromLocation(location), [location]);
+
+  const series = useMemo(() => {
+    const seriesMap: Record<string, SeriesDataUnit[]> = {};
+
+    for (const row of traces) {
+      const seriesName = row[colorEncoding];
+      if (!seriesMap.hasOwnProperty(seriesName)) {
+        seriesMap[seriesName] = [];
+      }
+      seriesMap[seriesName].push({
+        name: row.start_time_unix * 1000,
+        value: row.trace_duration_ms,
+      });
+    }
+
+    return Object.entries(seriesMap).map(([seriesName, data]) => ({seriesName, data}));
+  }, [colorEncoding]);
+
+  // TODO
+  const onClickHandler: EChartClickHandler = _params => {};
+
+  const loading = false;
+  const reloading = false;
+
+  const chartOptions = {
+    grid: {
+      left: '10px',
+      right: '10px',
+      top: '40px',
+      bottom: '0px',
+    },
+    tooltip: {
+      trigger: 'item' as const,
+      valueFormatter: (value: number) => tooltipFormatter(value, 'p50()'),
+    },
+    yAxis: {
+      axisLabel: {
+        color: theme.chartLabel,
+        formatter: (value: number) => axisLabelFormatter(value, 'p50()'),
+      },
+    },
+    legend: {
+      right: 10,
+      top: 5,
+      selected: getSeriesSelection(location),
+    },
+    onClick: onClickHandler,
+  };
+
+  return (
+    <Panel>
+      <ChartContainer>
+        <ChartZoom
+          router={router}
+          period={statsPeriod}
+          start={start}
+          end={end}
+          utc={utc === 'true'}
+        >
+          {zoomRenderProps => {
+            return (
+              <TransitionChart loading={loading} reloading={reloading}>
+                <TransparentLoadingMask visible={reloading} />
+                {getDynamicText({
+                  value: (
+                    <ScatterChart
+                      series={series}
+                      {...chartOptions}
+                      {...zoomRenderProps}
+                    />
+                  ),
+                  fixed: <Placeholder height="200px" />,
+                })}
+              </TransitionChart>
+            );
+          }}
+        </ChartZoom>
+      </ChartContainer>
+      <ChartControls>
+        <InlineContainer />
+        <InlineContainer>
+          <OptionSelector
+            title={t('Group By')}
+            selected={colorEncoding}
+            options={COLOR_ENCODINGS}
+            onChange={value => {
```

Comment:
I would React.useCallback this and name it handleColorEncodingChange which seems to be somewhat consistent w the rest of the codebase